### DISCUSSION TITLE
Tasmota Firmware Version Tracking

### DISCUSSION THREAD

**Discussion by tony-fav** [tony-fav/tasmota-blerry#30] [General]
Please report below you stability or instability with Tasmota32 firmware running Blerry.

Please include:
- Screenshot of the information page in Tasmota
- If self-compiled and stable, maybe share your user_config_override.h
- If pre-compiled, commit of dev branch it was from
- What physical device is being used (amazon link, aliexpress link, tindie, etc.), how is it powered?
- Steps taken (Erase Flash, flash new)
- Any SetOptions that have helped (`SO57 0` for example).
- If you were able to capture serial logging before failure or after failure, please linked to pastebin or something similar with the death logs.
- How many BLE devices AND which BLE devices (some require active scan and some do not, maybe active scan is the culprit, who knows??)

**Comment by monkey-of-the-hill** [comment]
Linkind Switch running tasmota32solo1-bluetooth.bin
Did a reset 5.
Ran blerry_setup_script.be
Pushed Config via mqtt using BlerrySetConfig command
so57 0
Current Uptime   1T07:13:06 (screen shot a bit older)
3 ATC sensors
![image](https://user-images.githubusercontent.com/4988077/150594816-4c39e631-7414-490e-b36c-4fd288ab9ec1.png)

**Comment by ElksInNC** [comment]
Linkind ESP32 Solo1 Switch flashed with Tasmota32 Solo1 Bluetooth
Web Flasher via serial and ERASE DEVICE selected.
Configured via serial console as follows:
- backlog devicename TESTBENCH; FriendlyName1 TESTBENCH; Hostname TESTBENCH; timezone 99; TimeDST 0,2,3,1,2,-240; TimeSTD 0,1,11,1,2,-300; SO8 1; SO53 1; SO55 1; SO57 0; ssid1 xxxxxx; password1 xxxxxx; 
- backlog template {"NAME":"Linkind Switch ESP32Solo WS240010008","GPIO":[0,0,0,0,0,224,1,1,0,0,320,0,1,1,0,0,0,0,0,0,0,576,289,0,0,0,0,0,33,32,0,0,0,0,0,0],"FLAG":0,"BASE":1}; module 0;
- backlog mqtthost 192.168.50.25; topic TESTBENCH; mqttuser xxxx; mqttpassword xxxxx; 

Further installed blerry v0.2 via web console with:
br import path; def start_blerry_setup(); var cl = webclient(); var url = 'https://raw.githubusercontent.com/tony-fav/tasmota-blerry/dev/blerry/blerry_setup.be'; cl.begin(url); var r = cl.GET(); if r != 200; print('error getting blerry_setup.be'); return false; end; var s = cl.get_string(); cl.close(); var f = open('blerry_setup.be', 'w'); f.write(s); f.close(); load('blerry_setup.be'); end; start_blerry_setup()

Finally installed blerry_config.json via console with 
BlerrySetConfig {"devices":{"D03232363E5D/1":{"alias":"Govee4Probe","model":"GVH5184","discovery":true,"via_pubs":true},"A4C138C5F025":{"alias":"MI-09","model":"ATCpvvx","discovery":true,"via_pubs":true},"A4C1383569C2":{"alias":"Evan-Bedroom","model":"ATCpvvx","discovery":true,"via_pubs":true},"A4C1380CC218":{"alias":"Server-Closet","model":"ATCpvvx","discovery":false},"A4C138EB4958":{"alias":"Upper-Attic","model":"ATCpvvx","discovery":false},"A4C1380B6A73":{"alias":"Upright-Freezer","model":"GVH5075","discovery":true},"A4C1388B8332":{"alias":"Mudroom-Lower-Freezer","model":"GVH5075","discovery":true},"A4C13838D005":{"alias":"Mudroom-Upper-Reefer","model":"ATCpvvx","discovery":true},"A4C138353F05":{"alias":"HPWH-Discharge","model":"ATCpvvx","discovery":false},"A4C138EB4397":{"alias":"Kitchen-Reefer","model":"ATCpvvx","discovery":false}},"override":{"base_topic":"tele/tasmota_blerry","discovery":true,"precision":{"Temperature":1,"Humidity":0,"Battery":-1}},"advanced":{"old_details":false}}

Operating conditions:
Card is powered via ESP-12 burner with power and serial lines connected via jumpers to pads on card.  This configuration has run successfully for 12+ days with no brownouts previously.  Solid power supply.

Seriallog 4 is on and all activity being logged.  Reset 99 to clear reboot count.  Updates if/when AP disconnect or reboot happens will be provided including complete serial log.

![image](https://user-images.githubusercontent.com/77821967/150601046-c9a1251a-8902-4d8a-a9ed-08e0bd00156b.png)

**Reply by ElksInNC** [reply]
For comparison - about 12 other switches all running 10.1.0.1 Thermostat (or a custom build from tony-fav of the same lineage) have been running strong with BLE scanning for 18+ days:
![image](https://user-images.githubusercontent.com/77821967/150612043-7c0d02b9-7afe-4784-aa13-1b35f517cb44.png)

![image](https://user-images.githubusercontent.com/77821967/150612157-d8580f10-5233-4a4b-bdfe-3bddd5c1a989.png)

**Reply by ElksInNC** [reply]
AP connection died at 18:54.  Start up time 16:15.

Last message sent to syslog also at 18:54.

Only message was Connect failed as AP cannot be reached.

```
18:54:03.234 WIF: Checking connection...
18:54:04.077 BRY: GC from 71794 to 61433 bytes, objects freed 90/1006 (in 8 ms)
18:54:04.092 BRY: GC from 71994 to 61782 bytes, objects freed 50/1006 (in 5 ms)
18:54:04.100 RSL: MI-09 = 7B2252535349223A2D33362C22416C [{"RSSI":-36,"Ali] ...
18:54:05.528 RSL: BLE = {"DetailsBLE":{"mac":"D03232363E5D/1","a":"Govee4Probe","RSSI":-50,"p":"0201060303518414FF363E5D01000101A801860834FFFF07FFFF1450"}}
18:54:06.459 BRY: GC from 72031 to 60945 bytes, objects freed 83/1009 (in 9 ms)
18:54:14.497 BRY: GC from 71186 to 61053 bytes, objects freed 143/1012 (in 8 ms)
18:54:23.145 WIF: Checking connection...
18:54:23.148 WIF: Connect failed as AP cannot be reached
18:54:28.514 WIF: Connecting to AP1 ElksInNC-IOT in mode 11n as TESTBENCH...
18:54:31.608 RSL: BLE = {"DetailsBLE":{"mac":"D03232363E5D/1","a":"Govee4Probe","RSSI":-44,"p":"0201060303518414FF363E5D01000101A801860834FFFF07FFFF1450"}}
18:54:31.646 BRY: GC from 71307 to 61745 bytes, objects freed 111/1019 (in 8 ms)
18:54:32.007 CFG: Saved, Count 28, Bytes 4096
18:54:32.159 WIF: Checking connection...
18:54:32.162 WIF: Connect failed as AP cannot be reached
```

Full logs from start until AP connect fail retained.  Serial Logging still running to see if device will panic/reboot.

**Reply by tony-fav** [reply]
That exact type of failure would be compensated for (solved* by the restart on WIFI disconnect rule. If it's just on a dev-board, I would probably do a `Restart 99` instead of `Restart 1` to prevent flash writes.

**Reply by ElksInNC** [reply]
Understood.  This is a test bench device that I'm intentionally let fail and see if I can catch any reboot death messages on serial log - so I do not have any rules in place.

In practice - if a disconnect/reboot rule keeps a production device rolling - why not.  But using a rule to reboot a device that keeps losing wifi is a bandaid, not a cure.  

Hopefully others who ARE trying to solve a production problem will start by rolling back to a 10.1.0.1 pre-dating the 2022.01.3 release as that seems to be the big problem right now.

**Reply by ElksInNC** [reply]
12 straight hours of this.  Giving up and rebooting manually.
```
07:43:34.999 WIF: Checking connection...
07:43:35.004 WIF: Connect failed as AP cannot be reached
07:43:35.998 WIF: Checking connection...
07:43:36.003 WIF: Connect failed as AP cannot be reached
07:43:41.379 WIF: Connecting to AP1 ElksInNC-IOT in mode 11n as TESTBENCH...
07:43:44.127 RSL: BLE = {"DetailsBLE":{"mac":"D03232363E5D/1","a":"Govee4Probe","RSSI":-53,"lost":true,"p":"0201060303518414FF363E5D01000101A801860834FFFF07FFFF1450"}}
07:43:44.947 WIF: Checking connection...
07:43:44.951 WIF: Connect failed as AP cannot be reached
07:43:45.944 WIF: Checking connection...
07:43:45.949 WIF: Connect failed as AP cannot be reached
```

**Reply by ElksInNC** [reply]
I wonder if this could be related or relevant?
https://github.com/espressif/arduino-esp32/issues/2501#issuecomment-1003746264

**Comment by devzwf** [comment]
TestBoard esp32  - ESP Chip Id  :  3226460 (ESP32-D0WDQ6)
Run Stock Tasmota32 bluetooth
{"SetOption57":"ON"}
5 x ATC
1 x Inkbird
1 x Govee
![esp32_blerry-Information](https://user-images.githubusercontent.com/25246962/150622072-eed0dd6d-b8f4-4580-8600-808b6ca0eb34.png)

**Reply by tony-fav** [reply]
By stock do you mean release 10.1.0?

**Reply by devzwf** [reply]
yes install from web-installer 
i mean i did not built it

**Comment by gmparis** [comment]
I have been using Blerry on three ESP32s for many months. They do require a restart every several weeks, but otherwise work fine if you don't add more than say 4 or 5 BLE devices to each. But now I am trying to build a new gateway using the exact same dev board. The only difference is the version of tasmota32-bluetooth. Sadly, Blerry does not work. Here's the error message:
`
BRY: failed to run compiled code (type_error - value 'nil' does not support subscriptable)
`
![image](https://github.com/tony-fav/tasmota-blerry/assets/24638881/86ebe91e-4c78-4966-adc7-fb465a35484d)

**Reply by tony-fav** [reply]
When in the logs does this happen? Right away?

**Reply by gmparis** [reply]
I reverted to the latest 13 version and it works, so I don't have it to try again. Here are more complete notes from when it was happening.
`
19:42:21.789 RUL: SYSTEM#BOOT performs 'br load('blerry.be')'
19:42:22.184 RSL: STATUS = {"Status":{"Module":1,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"tasmota_458078","ButtonTopic":"0","Power":"0","PowerLock":"0","PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
19:42:22.218 BRY: failed to run compiled code (type_error - value 'nil' does not support subscriptable)
19:42:22.230 RSL: RESULT = {"Br":"false"}
`
